src/fmt.cc

```diff
@@ -2,6 +2,9 @@ module;
 #ifndef __cpp_modules
 #  error Module not supported.
 #endif
+#ifdef _MSC_VER
+#  pragma warning(disable : 4702)
```

Comment:
What is this for?

Reply:
As mentioned in the commit message: it's about silencing warnings about 'unreachable code' bubbling up from the module when affected templates become instantiated in user code.

Reply:
Could you post the full warnings here? There might be better ways to suppress than using pragmas.

Reply:
From the compile log:
core.h(2668): warning C4702: unreachable code
core.h(2668): warning C4702: unreachable code
core.h(2668): warning C4702: unreachable code
core.h(2668): warning C4702: unreachable code
core.h(2657): warning C4702: unreachable code
core.h(2657): warning C4702: unreachable code
core.h(2668): warning C4702: unreachable code
core.h(2668): warning C4702: unreachable code
core.h(2668): warning C4702: unreachable code
core.h(2657): warning C4702: unreachable code
core.h(2657): warning C4702: unreachable code

I've checked: these pieces of unreachable code can be blessed with putting them into the `else` prongs of the respective `if constexpr ...` statements and the warnings vanish.

Reply:
I can do that even though it might look messy in the second instance.

Reply:
The code would look like 
```
constexpr int invalid_arg_index = -1;

#if FMT_USE_NONTYPE_TEMPLATE_PARAMETERS
template <int N, typename T, typename... Args, typename Char>
constexpr auto get_arg_index_by_name(basic_string_view<Char> name) -> int {
  if constexpr (detail::is_statically_named_arg<T>()) {
    if (name == T::name) return N;
  }
  if constexpr (sizeof...(Args) > 0) {
    return get_arg_index_by_name<N + 1, Args...>(name);
  } else { // **** the following 2 lines are moved into the additional else branch ****
    (void)name;  // Workaround an MSVC bug about "unused" parameter.
    return invalid_arg_index;
  }
  // **** lines were formerly here ****
}
#endif

template <typename... Args, typename Char>
FMT_CONSTEXPR auto get_arg_index_by_name(basic_string_view<Char> name) -> int {
#if FMT_USE_NONTYPE_TEMPLATE_PARAMETERS
  if constexpr (sizeof...(Args) > 0) {
    return get_arg_index_by_name<0, Args...>(name);
  } else { // **** additional else branch with verbatim copy ***
    (void)name;
    return invalid_arg_index;
  }
#else // **** formerly #endif ****
  (void)name;
  return invalid_arg_index;
#endif // *** formerly non-existent ****
}
```
The warning suppression can then be removed again and the code compiles without warnings at /W4 (i.e. STL warning level)

Reply:
It doesn't look too bad, let's change `get_arg_index_by_name` then.